ducks/transactions.ts

```diff
@@ -209,19 +197,53 @@ export function processTransaction(
 
     try {
       const actionCall = actions[unconfirmedTx.actionName as keyof typeof actions]
-      const response = await (actionCall as any)(tinlake, ...unconfirmedTx.actionArgs)
-      hasCompleted = true
+      const tx = await (actionCall as any)(tinlake, ...unconfirmedTx.actionArgs)
 
-      const outcome = (response as any).status === SUCCESS_STATUS
-      console.log(outcome)
-      outcomeTx.status = outcome ? 'succeeded' : 'failed'
-      outcomeTx.result = response
+      if (tx.hash) {
+        // Confirmed
+        const pendingTx: Transaction = {
+          ...unconfirmedTx,
+          status: 'pending',
+          hash: tx.hash,
+        }
+        await dispatch({ id, transaction: pendingTx, dontChangeUpdatedAt: true, type: SET_ACTIVE_TRANSACTION })
+
+        const pendingTxTimeout = 10000
+        const hidePendingTxCallback = async () => {
+          if (hasCompleted) return
+
+          if (!document.hidden) {
+            const hiddenPendingTx: Transaction = {
+              ...pendingTx,
+              showIfClosed: false,
+            }
+            await dispatch({
+              id,
+              transaction: hiddenPendingTx,
+              dontChangeUpdatedAt: true,
+              type: SET_ACTIVE_TRANSACTION,
+            })
+          } else {
+            setTimeout(hidePendingTxCallback, pendingTxTimeout)
+          }
+        }
 
-      if (errorMessageRegex.test(response.error)) {
-        const matches = response.error.toString().match(errorMessageRegex)
-        if (matches) outcomeTx.failedReason = matches[1]
-      } else if (outcomeTx.status === 'failed' && outcomeTx.result.message) {
-        outcomeTx.failedReason = outcomeTx.result.message
+        // Hide pending tx after 10s
+        setTimeout(hidePendingTxCallback, pendingTxTimeout)
+
+        const receipt = await tinlake.getTransactionReceipt(tx)
+        hasCompleted = true
+
+        const outcome = receipt.status === 1
+        outcomeTx.status = outcome ? 'succeeded' : 'failed'
+        outcomeTx.result = receipt
+        outcomeTx.hash = receipt.transactionHash
+      } else {
```

Comment:
This should never happen, right? If it can, could you add a comment here in which cases? If it can't, rather just fail with a console.error. Wdyt?

Reply:
This `else` happens if the transaction is for example rejected in Metamask by the user.

Reply:
There's a comment in the line below actually.